### PR TITLE
Add to updater support for uploading, column headers, more row info

### PR DESCRIPTION
**Pull request by fejta**
* Support rows that include metrics, messages, icons, cell ids.
* Support columns with extra headers.
* Marshal, compress and upload grid message
* Unit tests for metadata info

/assign @rmmh @michelle192837 

Many things this still does not do (sort by name, list builds in correct order, append metadata to row name, actually use the config to determine which builds to list, optimize for performance, run inside a container, etc...)

### CODE REVIEW

File: testgrid/cmd/updater/main.go [hunk 1]
```diff
@@ -392,6 +539,37 @@ func main() {
...
+	if err = zw.Close(); err != nil {
+		log.Fatalf("Failed to close zlib gs://%s/%s buffer: %v", b, o, err)
+	}
+	if b == "k8s-testgrid" {
+		log.Fatalf("do not change prod")
+	}
+	w := client.Bucket(b).Object(o).NewWriter(ctx)
+	w.SendCRC32C = true
+	buf = zbuf.Bytes()
+	w.ObjectAttrs.CRC32C = crc32.Checksum(buf, crc32.MakeTable(crc32.Castagnoli))
```
Comment (rmmh) [inline testgrid/cmd/updater/main.go]: Comment on why this is necessary. Also, maybe put that table in a global var block.

Comment (fejta) [reply testgrid/cmd/updater/main.go]: Done

**Comment by fejta** [comment]
/retest

mungegithub flake!!!

**Pull request merged**